decisiontree/forms/forms.py

```diff
@@ -0,0 +1,114 @@
+from django import forms
+from django.contrib.auth import get_user_model
+
+from decisiontree.multitenancy.forms import TenancyModelForm
+
+from .. import models
+from .fields import TagField
+
+
+class AnswerCreateUpdateForm(TenancyModelForm):
+
+    class Meta:
+        model = models.Answer
+        fields = ['name', 'type', 'answer', 'description']
+
+
+class AnswerSearchForm(forms.Form):
+    # ANALYSIS_TYPES = (
+    #     ('A', 'Mean'),
+    #     ('R', 'Median'),
+    #     ('C', 'Mode'),
+    # )
+    # answer = forms.ModelChoiceField(queryset=models.Answer.objects.none())
+    # analysis = forms.ChoiceField(choices=ANALYSIS_TYPES)
+    tag = forms.ModelChoiceField(
+        required=False, empty_label="All Tags",
+        queryset=models.Tag.objects.none())
+
+    def __init__(self, *args, **kwargs):
+        tree = kwargs.pop('tree')
+        super(AnswerSearchForm, self).__init__(*args, **kwargs)
+        # answers = models.Answer.objects.filter(transitions__entries__session__tree=tree)
+        tags = models.Tag.objects.filter(entries__session__tree=tree).distinct()
+
+        # self.fields['answer'].queryset = answers.distinct()
+        self.fields['tag'].queryset = tags
+        # self.fields['analysis'].label = 'Calculator'
+        # self.fields['tag'].label = 'Calculator'
```

Comment:
I suggest that we keep them around for a little while. The answer search form actually isn't hooked up properly (it renders but filling it out has no effect), but I haven't analyzed it fully to see how it could/should work.